crates/isograph_schema/src/validate_entrypoint.rs

```diff
@@ -1,34 +1,76 @@
-use std::collections::HashMap;
+use std::collections::{hash_map::Entry, HashMap};
+use std::fmt;
 
 use common_lang_types::{
     IsoLiteralText, IsographObjectTypeName, Location, ServerScalarSelectableName, TextSource,
     UnvalidatedTypeName, WithLocation, WithSpan,
 };
 use isograph_lang_types::{
-    ClientScalarSelectableId, DefinitionLocation, EntrypointDeclaration, SelectionType,
-    ServerEntityId, ServerObjectEntityId,
+    ClientScalarSelectableId, DefinitionLocation, EntrypointDeclaration, EntrypointDirectiveSet,
+    SelectionType, ServerEntityId, ServerObjectEntityId,
 };
 
 use thiserror::Error;
 
 use crate::{NetworkProtocol, Schema};
 
+#[derive(Debug)]
+pub struct LoadTypedIsoLiteralText {
+    pub iso_literal_text: IsoLiteralText,
+    pub is_lazy_loaded: bool,
+}
+
+impl LoadTypedIsoLiteralText {
+    pub fn new(entrypoint_declaration: WithSpan<EntrypointDeclaration>) -> Self {
+        let lazy_status = matches!(
+            entrypoint_declaration.item.entrypoint_directive_set,
+            EntrypointDirectiveSet::LazyLoad(_)
+        );
+        LoadTypedIsoLiteralText {
+            iso_literal_text: entrypoint_declaration.item.iso_literal_text,
+            is_lazy_loaded: lazy_status,
+        }
+    }
+}
+
+impl fmt::Display for LoadTypedIsoLiteralText {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.iso_literal_text)
+    }
+}
+
 pub fn validate_entrypoints<TNetworkProtocol: NetworkProtocol>(
     schema: &Schema<TNetworkProtocol>,
     entrypoint_declarations: Vec<(TextSource, WithSpan<EntrypointDeclaration>)>,
 ) -> Result<
-    HashMap<ClientScalarSelectableId, IsoLiteralText>,
+    HashMap<ClientScalarSelectableId, LoadTypedIsoLiteralText>,
     Vec<WithLocation<ValidateEntrypointDeclarationError>>,
 > {
     let mut errors = vec![];
-    let mut entrypoints = HashMap::new();
+    let mut entrypoints: HashMap<ClientScalarSelectableId, LoadTypedIsoLiteralText> =
+        HashMap::new();
     for (text_source, entrypoint_declaration) in entrypoint_declarations {
         match validate_entrypoint_type_and_field(schema, text_source, entrypoint_declaration) {
             Ok(client_field_id) => {
-                entrypoints.insert(
-                    client_field_id,
-                    entrypoint_declaration.item.iso_literal_text,
-                );
+                let new_entrypoint = LoadTypedIsoLiteralText::new(entrypoint_declaration);
+                match entrypoints.entry(client_field_id) {
+                    Entry::Occupied(occupied_entry)
+                        if (occupied_entry.get().is_lazy_loaded
```

Comment:
let's do `Entry::Occupied(occupied_entry) { if (...) { ... }  }` and let's not do `_` but do `Entry::Vacant(vacant) => vacant.insert(...)`